ansible/roles/test/tasks/bgp_speaker.yml

```diff
@@ -0,0 +1,153 @@
+#========================================
+# Run BGP Speaker test
+#========================================
+
+- fail: msg="Information about tested missing"
+  when: (testbed_type is not defined)
+
+- fail: msg="Invalid testbed_type value '{{testbed_type}}'"
+  when: testbed_type not in ['t0']
+
+- include_vars: "vars/deployment_id_asn_map.yml"
+
+- name: Gather minigraph facts about the device
+  minigraph_facts: host={{inventory_hostname}}
+  become: no
+  connection: local
+
+- name: print deployment id
+  debug: msg="{{ deployment_id }}"
+
+- name: set bgp speaker asn number
+  set_fact:
+     bgp_speaker_asn={{deployment_id_asn_map[deployment_id]}}
```

Comment:
be consistent on the space, in line 19, you have space between braces and word. here you do not.